File: quickstarts/CSharp/Speak.cs
using System;
using System.IO;
using System.Net;

namespace TranslateTextQuickStart
{
    class Program
    {
        static string host = "https://api.microsofttranslator.com";
        static string path = "/V2/Http.svc/Speak";

        // NOTE: Replace this example key with a valid subscription key.
        static string key = "ENTER KEY HERE";

        static void Speak()
        {
            string text = "Hello world";
            string language = "en-US";
            string output_path = "speak.wav";

            string uri = host + path + "?text=" + System.Net.WebUtility.UrlEncode(text) + "&language=" + language;

            WebRequest webRequest = WebRequest.Create(uri);
            webRequest.Headers.Add("Ocp-Apim-Subscription-Key", key);

            using (WebResponse response = webRequest.GetResponse())
            using (Stream stream = response.GetResponseStream())
            using (var fileStream = File.Create(output_path))
            {
                stream.CopyTo(fileStream);
                Console.WriteLine("File written.");
            }
        }

        static void Main(string[] args)
        {
            Speak();
            Console.ReadLine();
        }
    }
}

File: quickstarts/PHP/Speak.php
<?php

// NOTE: Be sure to uncomment the following line in your php.ini file.
// ;extension=php_openssl.dll

// **********************************************
// *** Update or verify the following values. ***
// **********************************************

// Replace the subscriptionKey string value with your valid subscription key.
$key = 'ENTER KEY HERE';

$host = "https://api.microsofttranslator.com";
$path = "/V2/Http.svc/Speak";

$text = "Hello world";
$language = "en-US";
$output_path = "speak.wav";

$params = "?text=" . urlencode ($text) . "&language=$language";

function Speak ($host, $path, $key, $params) {

	$headers =
		"Content-type: text/xml\r\n" .
		"Ocp-Apim-Subscription-Key: $key\r\n";

	// NOTE: Use the key 'http' even if you are making an HTTPS request. See:
	// http://php.net/manual/en/function.stream-context-create.php
	$options = array (
		'http' => array (
			'header' => $headers,
			'method' => 'GET'
		)
	);
	$context  = stream_context_create ($options);
	$result = file_get_contents ($host . $path . $params, false, $context);
	return $result;

}

$result = Speak ($host, $path, $key, $params);

$out = fopen($output_path, 'w');
// NOTE: fwrite is binary-safe. See: http://php.net/manual/en/function.fwrite.php
fwrite($out, $result);

echo "File written.";
?>
